agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/configuration/ConnectionStringBuilder.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.azure.monitor.opentelemetry.exporter.implementation.configuration;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+class ConnectionStringBuilder {
+
+  private URL ingestionEndpoint;
+  private URL liveEndpoint;
+  private URL profilerEndpoint;
+  private URL snapshotEndpoint;
+
+  ConnectionStringBuilder() {
+    try {
+      ingestionEndpoint = new URL(DefaultEndpoints.INGESTION_ENDPOINT);
+      liveEndpoint = new URL(DefaultEndpoints.LIVE_ENDPOINT);
+      profilerEndpoint = new URL(DefaultEndpoints.PROFILER_ENDPOINT);
+      snapshotEndpoint = new URL(DefaultEndpoints.SNAPSHOT_ENDPOINT);
+    } catch (MalformedURLException e) {
+      throw new IllegalStateException("ConnectionString.Defaults are invalid", e);
+    }
+  }
+
+  URL getIngestionEndpoint() {
+    return ingestionEndpoint;
+  }
+
+  void setIngestionEndpoint(String ingestionEndpoint) {
```

Comment:
nit: instead of setters can you please change this to builderpattern conventions and also add a build function instead of default constructor